library/std/src/os/mod.rs

```diff
@@ -178,7 +180,7 @@ pub mod vxworks;
 #[cfg(target_os = "xous")]
 pub mod xous;
 
-#[cfg(any(unix, target_os = "hermit", target_os = "wasi", doc))]
+#[cfg(any(unix, target_os = "hermit", target_os = "trusty", target_os = "wasi", doc))]
```

Comment:
This whole module is about providing safe abstractions for file descriptors that maintain IO safety. But if I understand the documentation correctly, there is no way to open or close file descriptors on Trusty, meaning the whole concept of IO safety is unnecessary. Thus I don't think that this module should exist on Trusty.

Reply:
We do make heavy use of `OwnedFd` in Trusty. While userspace applications can't directly open e.g. a file, we do have a [concept of handles](https://source.android.com/docs/security/features/trusty/trusty-ref#handle_api) that are used for things like IPC. We'd also like to have access to the `os::fd` module because Trusty shares code with Android, and it'd be difficult if we had two different fd types. Sharing the same fd infrastructure in libstd simplifies things for us a lot.